libraries/chain/asset_evaluator.cpp

```diff
@@ -559,24 +559,12 @@ static bool update_bitasset_object_options(
       const auto old_feed = bdo.current_feed;
       bdo.update_median_feeds( db.head_block_time(), next_maint_time );
 
-      // TODO review and refactor / cleanup after hard fork:
-      //      1. if hf_core_868_890 and core-935 occurred at same time
-      //      2. if wlog did not actually get called
-
-      // We need to call check_call_orders if the price feed changes after hardfork core-935
-      if( next_maint_time > HARDFORK_CORE_935_TIME )
-         return ( !( old_feed == bdo.current_feed ) );
-
       // We need to call check_call_orders if the settlement price changes after hardfork core-868-890
       if( after_hf_core_868_890 )
       {
-         if( old_feed.settlement_price != bdo.current_feed.settlement_price )
+         if( old_feed.settlement_price != bdo.current_feed.settlement_price ||
+               !( old_feed == bdo.current_feed ) )
```

Comment:
Better directly `return ( !( old_feed == bdo.current_feed ) );`? Because `old_feed.settlement_price != bdo.current_feed.settlement_price` is a subset of `!( old_feed == bdo.current_feed )`.

Reply:
I left that in with the intent to evaluate the == operator of the feed later. And then I promptly forgot. It has been fixed.